src/sql-parser/src/parser.rs

```diff
@@ -4399,9 +4421,44 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_option_map<T, F>(
+        &mut self,
+        mut f: F,
+    ) -> Result<Option<BTreeMap<String, T>>, ParserError>
+    where
+        F: FnMut(&mut Self) -> Result<T, ParserError>,
+    {
+        Ok(if self.parse_keyword(MAP) {
+            self.expect_token(&Token::LBracket)?;
+            let mut map: BTreeMap<String, T> = BTreeMap::new();
+            loop {
+                if let Some(Token::RBracket) = self.peek_token() {
+                    break;
+                }
+                let key = match self.next_token() {
+                    Some(Token::String(s)) => s,
+                    Some(Token::Ident(s)) => s.into_inner(),
```

Comment:
Preference to just keep this to strings. Option values have to conflate idents and strings for PostgreSQL compatibility, but we can avoid that wart in map keys at least.